cpp/src/NVStrings_array.cu

```diff
@@ -132,6 +139,44 @@ NVStrings* NVStrings::gather( const bool* mask, bool bdevmem )
     return gather( d_indexes, count, true );
 }
 
+//
+// s1 = ['a','b,'c','d']
+// s2 = ['e','f']
+// pos = [1,3]  -- must be the same length as s2
+// s3 = s1.scatter(s2,pos)
+// ['a','e','c','f']
+//
+NVStrings* NVStrings::scatter( NVStrings& strs, const int* pos, bool bdevmem )
+{
+    unsigned int count = size();
+    unsigned int elements = strs.size();
+    if( pos==0 )
+        throw std::invalid_argument("position parameter cannot be null");
+
+    auto execpol = rmm::exec_policy(0);
+    const int* d_pos = pos;
+    if( !bdevmem )
+    {   // copy indexes to device memory
+        d_pos = const_cast<const int*>(device_alloc<int>(elements,0));
+        CUDA_TRY(cudaMemcpyAsync((void*)d_pos,pos,elements*sizeof(int),cudaMemcpyHostToDevice))
```

Comment:
It might be a good idea to have an explicit check for strs.size()==len(indexes) at the python level or someplace else. I tried some experiments with providing indexes smaller than strs.size() and looking at the logic it seems like we might try to access memory that is not owned by pos. (Both when pos is on host and on device)

Reply:
If this is a plain device pointer, this will not be possible.

Reply:
Added check for non-plain device pointer types in `pystrings.cpp`.